docs/DeveloperGuide.md

```diff
@@ -1,38 +1,516 @@
+having one `deleteItem` method take in an `Integer` and another taking in a `String`.
+* If the parameter is an `Integer`, the `deleteItem(int index)` method will call the `remove` method of the `ArrayList` class to
+remove the item from the inventory list.
+* If the parameter is a `String`, the `deleteItem(String keyword)` method will run a `for` loop to iterate through the `ArrayList`
+until it finds a `Item` object whose name `equals` to that of the search parameter. If an `Item` object has matching
+names with the search parameter, it will store the index in the `targetIndex` variable. This `deleteItem(String keyword)` method will
+then call another `deleteItem(int index)` method, but this time, the parameter passed is an integer. The execution after this
+will be exactly the same as passing an `Integer` to the `deleteItem(int index)` method mentions above.
+
+Upon completion of either operation, the `execute()` method sets the `hasToSave` flag to true, signaling the need to persist changes to storage. This entire process is logged at various levels, providing a trail for audit and debugging purposes.
+
+#### Implementation Notes ####
+The design of `DeleteCommand` is such that it encapsulates the delete operation, separating concerns by handling the logic of deletion internally and interacting with `ItemList` without exposing the details of the list's data structure. This modular approach simplifies the responsibilities of the `Parser` and `BinBash` (main) components, which do not need to understand the internals of the delete operation, thereby adhering to the principles of high cohesion and low coupling.
+
+Additionally, the decision to use two constructors promotes the Single Responsibility Principle, as each constructor's logic is tailored to the type of deletion it handles.
+
+### Search command
+
+![SearchAssistantSequenceDiagram](images/SearchAssistantSequenceDiagram.png)
+The search command calls on methods in the SearchAssistant class to perform queries.
+
+![SearchAssistantReferenceFrameDiagram](images/SearchAssistantReferenceFrameDiagram.png)
+Each method searches through a corresponding field.
+If the argument to this method is found to be a certain default value (MIN/MAX values for numerical arguments and empty strings for string arguments), searching through this field is skipped.
+
+Note that search relies on the SearchAssistant having the full item list to produce correct results.
+This is guaranteed in this case by ItemList's setFoundItems() call.
+
+After obtaining the list of found items, it then uses printList to convert this list into a user friendly string.
+
+## Logging
+
+Logging plays a crucial role in providing insights into the runtime behavior of the application and diagnosing issues. In this project, we utilize the `java.util.logging` package for comprehensive logging functionality.
+
+### BinBashLogger Class
```

Comment:
While logging is mentioned, maybe a dedicated section on common errors users might encounter, and troubleshooting steps could improve user experience, especially for new users unfamiliar with CLI applications.